### ISSUE TITLE
scalac gives noisey warnings with small compile classpaths

### ISSUE THREAD

**Issue by johnynek** [bazelbuild/rules_scala#10]
See: https://issues.scala-lang.org/browse/SI-9673

For instance:
```
07:09:29 warning: Class org.apache.commons.logging.Log not found - continuing with a stub.
07:09:29 one warning found
which will be added to the runtime classpath, but is indeed not on the compile time classpath.
```

I don't see anything we can do here but hope scalac improves.

**Comment by johnynek** [comment]
I added this:

https://github.com/scala/scala/pull/5268

and am using an internal build of that which solves the issue. Maybe we can get it into mainline.

**Comment by johnynek** [comment]
This was merged into mainline in 2.12:

https://github.com/scala/scala/pull/5297

I may send a PR for 2.11.9 or maybe the typelevel compiler, but for now, I'm closing.
